webapi/src/test/scala/org/knora/webapi/slice/common/repo/rdf/RdfModelSpec.scala

```diff
@@ -0,0 +1,1065 @@
+/*
+ * Copyright © 2021 - 2024 Swiss National Data and Service Center for the Humanities and/or DaSCH Service Platform contributors.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.knora.webapi.slice.common.repo.rdf
+
+import org.apache.jena.riot.RiotException
+import zio.*
+import zio.test.Assertion.*
+import zio.test.*
+
+import org.knora.webapi.slice.common.repo.rdf.Errors.ConversionError
+import org.knora.webapi.slice.common.repo.rdf.Errors.LiteralNotPresent
+import org.knora.webapi.slice.common.repo.rdf.Errors.NotALiteral
+import org.knora.webapi.slice.common.repo.rdf.Errors.ObjectNotAResource
+import org.knora.webapi.slice.common.repo.rdf.Errors.ObjectNotPresent
+import org.knora.webapi.slice.common.repo.rdf.Errors.ResourceNotPresent
+import org.knora.webapi.slice.resourceinfo.domain.InternalIri
+
+object RdfModelSpec extends ZIOSpecDefault {
+
+  private case class StringContainer(value: String)
+  private implicit val stringContainerConverter: String => Either[String, StringContainer] =
+    str => Right(StringContainer(str))
+
+  private case class FailingStringContainer(value: String)
+  private implicit val failingStringContainerConverter: String => Either[String, FailingStringContainer] =
+    str => Left("Conversion failed")
+
+  private case class LangStringContainer(value: LangString)
+  private implicit val langStringContainerConverter: LangString => Either[String, LangStringContainer] =
+    langString => Right(LangStringContainer(langString))
+
+  private case class FailingLangStringContainer(value: LangString)
+  private implicit val failingLangStringContainerConverter: LangString => Either[String, FailingLangStringContainer] =
+    langString => Left("Conversion failed")
+
+  private case class BooleanContainer(value: Boolean)
+  private implicit val booleanContainerConverter: Boolean => Either[String, BooleanContainer] =
+    bool => Right(BooleanContainer(bool))
+
+  private case class FailingBooleanContainer(value: Boolean)
+  private implicit val failingBooleanContainerConverter: Boolean => Either[String, FailingBooleanContainer] =
+    bool => Left("Conversion failed")
+
+  private val rdfModelSuite = suite("RdfModel")(
+    suite("Deserialize turtle")(
+      test("Create an RdfModel from valid turtle") {
+        val turtle =
+          """|@prefix ex: <http://example.org/> .
+             |ex:subject ex:predicate ex:object .
+             |""".stripMargin
+        for {
+          rdfModel <- RdfModel.fromTurtle(turtle).exit
+        } yield assertTrue(rdfModel.isSuccess)
+      },
+      test("Fail to create an RdfModel from invalid turtle") {
+        for {
+          rdfModel <- RdfModel.fromTurtle("Not turtle").exit
+        } yield assert(rdfModel)(failsWithA[RiotException])
```

Comment:
We should think about wrapping also the `RiotException` into our own error type.
This is leaking implementation detail of jena.

Reply:
you're right